libraries/chain/block_log.cpp

```diff
@@ -279,7 +281,10 @@ namespace eosio { namespace chain {
    public:
 
      block_log_data() = default;
-     block_log_data(const fc::path& path, mapmode mode = mapmode::readonly) { open(path, mode); }
+     block_log_data(const fc::path& path, mapmode mode = mapmode::readonly) { 
+        preamble.log_path = path;
+        open(path, mode); 
+     }
 
      const block_log_preamble& get_preamble() const { return preamble; }
 
```

Comment:
`open` is public but does not set the `log_path`
Should either be made private or set `log_path`.

Reply:
It is in constructor, preamble is struct

Reply:
But if you call it `open` again with a diff log_path they will be out of sync.

Reply:
I see the point, thanks, it is better to update in open also.